### PR TITLE
Adding metafunction to correct type used in buffer_allocator when dev…

### PR DESCRIPTION
**Pull request by timmiesmith**
…ice_pointer value type is void.

### CODE REVIEW

**Comment by timmiesmith** [comment]
@ShengchenJ please review.

File: clang/runtime/dpct-rt/include/dpl_extras/memory.h.inc [hunk 1]
```diff
@@ -50,13 +50,31 @@
 // DPCT_LABEL_END
 namespace dpct {
 namespace detail {
-// DPCT_LABEL_BEGIN|__buffer_allocator|dpct::detail
+template <typename T>
+// DPCT_LABEL_BEGIN|make_allocatable|dpct::detail
 // DPCT_DEPENDENCY_EMPTY
```
Comment (ShengchenJ) [inline clang/runtime/dpct-rt/include/dpl_extras/memory.h.inc]: The code depends on the dpct::byte_t, please help update it.

**Review by ShengchenJ — CHANGES_REQUESTED** [review]
The patch is LGTM, but the Lit test failed, the check number is updated. Could you update the lit cases? Many thanks.

File: clang/runtime/dpct-rt/include/dpl_extras/memory.h.inc (hunk 1 above)
Comment (timmiesmith) [reply clang/runtime/dpct-rt/include/dpl_extras/memory.h.inc]: Dependence information has been updated.

**Comment by zhiweij1** [comment]
I have merged https://github.com/oneapi-src/SYCLomatic/pull/316 based on this one and fixed all LIT failures.

**Pull request closed without merging**